### PR TITLE
explain how to run builder-web and get faster builds in a linux vm

### PR DESCRIPTION
**Pull request by mwrock**
This includes some clarifications around running the depot web service and slow build times over a synced network folder that should hopefully help others.

Signed-off-by: Matt Wrock <matt@mattwrock.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by mwrock** [comment]
Note that #2108 is necessary for some of the oauth instructions to be true. Otherwise, db user, server and name need to be added to the config.toml files

**Comment by adamhjk** [comment]
@mwrock should we just start builder-web with the other services?

**Comment by adamhjk** [comment]
this looks good to me!

gif-keyboard-6094338424868465305

**Comment by adamhjk** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/222847193).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/222847193).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: components/builder-web/README.md [hunk 1]
```diff
@@ -21,11 +13,78 @@ The configuration file looks like:
...
+
+## Github OAuth
+
+By default, `builder-web` is configured to use a preconfigured dev github oauth application. This should suffice as long as you intend to use `http://localhost:3000` as the homepage. If you need to use an alternate host name or port, you will need to setup a separate oauth application and configure `builder-api` and `builder-sessionsrv` with its generated credentials.
+
+To register a new oauth application, go to your github user account settings and navigate to `OAuth Applications` and then click on `Register a new application`.
+
+It is important that the homepage is set to `http://<hostname>:<port>` and the Authorization callback URL is set to `http://<hostname>:<port>/#/sign-in`.
+
+Set the `github_client_id` to the client ID assigned to the oauth application. If you are running the API services, add `cvonfig.toml` files for the `builder-sessionsrv` and `builder-api` services:
```
Comment (raskchanky) [inline components/builder-web/README.md]: This should be `config.toml`, not `cvonfig.toml`, yes?

**Comment by mwrock** [comment]
oops. I'll edit and push @raskchanky thanks! @adamhjk I thought about that too or at least adding another make command that added it. Maybe I'll add that today. Just need to add a support script that does the node stuff.

**Pull request merged**